src/components/Books.js

```diff
@@ -1,14 +1,23 @@
-import { useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchBooks } from '../redux/ApiCalls';
+import { setBooks } from '../redux/books/books';
 import AddBook from './AddBook';
 import SingleBook from './SingleBook';
 
 const Books = () => {
   const books = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+
+  useEffect(async () => {
+    const data = await fetchBooks();
+    dispatch(setBooks(data));
+  }, []);
 
   return (
     <div>
       <ul className="list-group">
-        {books.length > 0 ? books.map((book) => <SingleBook key={book.id} book={book} />)
+        {books.length > 0 ? books.map((book) => <SingleBook key={book.item_id} book={book} />)
```

Comment:
- [ ] There are some errors related to duplicate keys when rendering these books. I guess it can be due to the response object parsing that you have done in the apiCalls files but you need to have a look at that and fix it.

Here is the error: 
Reproduce the error: 
1 - Click on the categories link
2 - Click on the Books link

![Screenshot from 2021-09-28 16-15-53](https://user-images.githubusercontent.com/40503704/135116843-60667d9b-350f-4153-b8c2-1ee0bce468e4.png)